### PR TITLE
Floating Messages

### PR DESCRIPTION
**Pull request by Geevies**
* Added optional floating messages above people's heads when they speak. Press Toggle Floating Messages in the OOC tab to enable this.

Ported from https://github.com/NebulaSS13/Nebula/pull/349, who got it from https://github.com/BeeStation/BeeStation-Hornet/pull/1603, which got it from Goon and Burgerstation (I think). Burger had a big part in making this an SS13 thing.

No medbots were harmed in the acquisition of this screenshot.
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by CthulhuOnIce** [comment]
😳

File: code/modules/mob/floating_messages.dm [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	if(small)
+		fontsize = 5
+	if(copytext(message, length(message) - 1) == "!!")
+		fontsize = 8
+		style += "font-weight: bold;"
+
+	if(length(message) > 50)
+		message = "[copytext(message, 1, 50)]..."
+
+	style += "color: [get_floating_message_color(name)];"
```
Comment (CthulhuOnIce) [inline code/modules/mob/floating_messages.dm:20]: wouldn't this be kind of expensive to run every time someone says something

Comment (Geevies) [reply code/modules/mob/floating_messages.dm:20]: no idea, someone smarter than me will probably think of something better, but i doubt this is too intense

Comment (Geevies) [reply code/modules/mob/floating_messages.dm:20]: we have a whole 32 bits to deal with it, lmfao

**Pull request merged**